web/themes/new_weather_theme/assets/js/components/DailyForecast.js

```diff
@@ -0,0 +1,168 @@
+/**
+ * Daily Forecast Component
+ * -------------------------
+ * This component wraps the main elements of the
+ * Daily Forecast tabpanel, specifically:
+ * - the quick forecast nav / tab component
+ * - the list containing daily forecasts
+ * Its purpose is to manage setting the daily forecast into
+ * "quick forecast tab mode" (on desktop) vs "quick toggle"
+ * mode.
+ * For a11y purposes, we need to add and remove aria attributes
+ * and roles dynamically when breaking to and from desktop width.
+ * This component also overrides click behavior on the quick
+ * forecast nav links, forcing them to operate as tabs that
+ * hide/show relevant forecast day list items.
+ */
+class DailyForecast extends HTMLElement {
+  constructor(){
+    super();
+
+    // Bound methods
+    this.tabClickHandler = this.tabClickHandler.bind(this);
+    this.setupMediaEvents = this.setupMediaEvents.bind(this);
+    this.setupTabMode = this.setupTabMode.bind(this);
+    this.undoTabMode = this.undoTabMode.bind(this);
+    this.handleDesktopMediaChange = this.handleDesktopMediaChange.bind(this);
+  }
+
+  connectedCallback(){
+    this.setupMediaEvents();
+  }
+
+  disconnectedCallback(){
+    Array.from(
+      this.querySelectorAll(".wx-quick-forecast-item")
+    ).forEach(item => {
+      item.removeEventListener("click", this.tabClickHandler);
+    });
+  }
+
+  /**
+   * In order to respond to the change in desktop
+   * breakpoint / media query, we need to pull out
+   * preset CSS Custom Property values and then
+   * set up matchMedia listeners.
+   * This method does so, and also calls the handler
+   * immediately, so as to setup the component for
+   * the given media (desktop or everything else)
+   */
+  setupMediaEvents(){
+    const computedStyle = getComputedStyle(this);
+    this.desktopBreakpoint = computedStyle.getPropertyValue("--wx-media-breakpoint-desktop");
+    this.desktopQuery = window.matchMedia(`(min-width: ${this.desktopBreakpoint})`);
+    this.desktopQuery.addEventListener("change", this.handleDesktopMediaChange);
+    this.handleDesktopMediaChange();
+  }
```

Comment:
The actual measurements for the `desktop` breakpoint in our USDWS setup get pulled from the CSS custom property here. This is also where we configure the `matchMedia` object and set up a function to react to changes on it. That ends up being a listener for when the desktop breakpoint changes.